### PR TITLE
Migrate metadata of QueryCoord from 2.1 to 2.2

### PR DESCRIPTION
**Pull request by yah01**
Signed-off-by: yah01 <yang.cen@zilliz.com>
/kind improvement
related #19960

### CODE REVIEW

File: cmd/tools/migration/backend/etcd210.go [hunk 1]
```diff
@@ -293,6 +295,35 @@ func (b etcd210) loadLastDDLRecords() (meta.LastDDLRecords, error) {
 	return records, nil
 }
 
+func (b etcd210) loadLoadInfos() (meta.CollectionLoadInfo210, error) {
+	loadInfo := make(meta.CollectionLoadInfo210)
+	_, collectionValues, err := b.txn.LoadWithPrefix(legacy.CollectionMetaPrefixV1)
```
Comment (longjiquan) [inline cmd/tools/migration/backend/etcd210.go:300]: Maybe `CollectionLoadMetaPrefixV1` is better.

File: cmd/tools/migration/backend/etcd210.go [hunk 2]
```diff
@@ -293,6 +295,35 @@ func (b etcd210) loadLastDDLRecords() (meta.LastDDLRecords, error) {
...
+	if err != nil {
+		return nil, err
+	}
+	for _, value := range collectionValues {
+		collectionInfo := querypb.CollectionInfo{}
+		err = proto.Unmarshal([]byte(value), &collectionInfo)
+		if err != nil {
+			return nil, err
+		}
+		if collectionInfo.InMemoryPercentage < 100 {
```
Comment (longjiquan) [inline cmd/tools/migration/backend/etcd210.go:310]: Why we ignore these unloaded collections? One quick question, if the users called the `load` before query service is down, should querycoord recover these collections anyway?

File: cmd/tools/migration/meta/210_to_220.go [hunk 3]
```diff
@@ -176,6 +210,31 @@ func combineToSegmentIndexesMeta220(segmentIndexes SegmentIndexesMeta210, indexB
 	return segmentIndexModels, nil
 }
 
+func combineToLoadInfo220(collectionLoadInfo CollectionLoadInfo220, partitionLoadInto PartitionLoadInfo220, fieldIndexes FieldIndexes210) {
+	for collectionID, loadInfo := range collectionLoadInfo {
+		indexes, ok := fieldIndexes[collectionID]
+		if !ok {
+			delete(collectionLoadInfo, collectionID)
```
Comment (longjiquan) [inline cmd/tools/migration/meta/210_to_220.go:217]: I think this behavior needs more discussion. Should we warn here or fill an flat index meta? My concern is that if users loaded collections without index in the earlier version, in fact we break the compatiblity here and the users need to re-index the collection and then reload the collection.

File: cmd/tools/migration/meta/210_to_220.go [hunk 4]
```diff
@@ -176,6 +210,31 @@ func combineToSegmentIndexesMeta220(segmentIndexes SegmentIndexesMeta210, indexB
...
+
+		for _, index := range indexes.indexes {
+			loadInfo.FieldIndexID[index.GetFiledID()] = index.GetIndexID()
+		}
+	}
+
+	for collectionID, partitions := range partitionLoadInto {
+		indexes, ok := fieldIndexes[collectionID]
+		if !ok {
+			delete(collectionLoadInfo, collectionID)
```
Comment (longjiquan) [inline cmd/tools/migration/meta/210_to_220.go:228]: Ditto.

**Review by longjiquan — CHANGES_REQUESTED** [review]

File: cmd/tools/migration/backend/etcd210.go (hunk 2 above)
Comment (yah01) [reply cmd/tools/migration/backend/etcd210.go:310]: since v2.2, the loading collection will be released after restart

**Review by congqixia — COMMENTED** [review]
/approve

**Pull request merged**